source/FreeRTOS_Sockets.c

```diff
@@ -2928,6 +2934,24 @@ BaseType_t FreeRTOS_setsockopt( Socket_t xSocket,
                         xReturn = prvSetOptionStopRX( pxSocket, pvOptionValue );
                         break;
                 #endif /* ipconfigUSE_TCP == 1 */
+                #if ( ipconfigPACKET_PRIORITIES > 1 )
+                    case FREERTOS_SO_PRIORITY:
+                       {
+                           uint8_t * pucValue = ( uint8_t * ) pvOptionValue;
```

Comment:
In Linux, the priority is passed as an `int`, in FreeRTOS+tCP we have the habit of using `BaseType_t`:
~~~diff
-    uint8_t * pucValue = ( uint8_t * ) pvOptionValue;
+    BaseType_t * pxValue = ( BaseType_t * ) pvOptionValue;
~~~